File: frontend/src/types/order.ts
export interface OrderItem {
  product_id: number;
  quantity: number;
  unit_price: number;
  total_price: number;
}

export interface Order {
  id: number;
  customer_id: number;
  shipping_method_id: number;
  total_amount: number;
  status: string;
  created_at: string;
  items: OrderItem[];
}

File: backend/app/schemas/order.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class OrderItemBase(BaseModel):
    product_id: int
    quantity: int
    unit_price: float

class OrderItemCreate(OrderItemBase):
    pass

class OrderItemOut(OrderItemBase):
    id: int

    class Config:
        from_attributes = True


class OrderBase(BaseModel):
    customer_id: int
    total_amount: float
    status: Optional[str] = "Completed"

class OrderCreate(OrderBase):
    items: List[OrderItemCreate]

class OrderOut(OrderBase):
    id: int
    created_at: datetime
    items: List[OrderItemOut] = []

    class Config:
        from_attributes = True